docs/extensions/logging-and-monitoring.md

```diff
@@ -132,17 +132,203 @@ Examples for this would be a cloud-controller-manager or CSI controller deployme
 
 #### Extensions Monitoring Integration
 
+In case an extension wants to extend the configuration for the shoot Prometheus, they can create the [`prometheus-operator`'s custom resources](https://github.com/prometheus-operator/prometheus-operator?tab=readme-ov-file#customresourcedefinitions) and label them with `prometheus=shoot`.
+
+##### `ServiceMonitor`
+
+When the component runs in the seed cluster (e.g., as part of the shoot control plane), `ServiceMonitor` resources should be used:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    prometheus: shoot
+  name: shoot-my-component
+  namespace: shoot--foo--bar
+spec:
+  selector:
+    matchLabels:
+      app: my-component
+  endpoints:
+  - metricRelabelings:
+    - action: keep
+      regex: ^(metric1|metric2|...)$
+      sourceLabels:
+      - __name__
+    port: metrics
+```
+
+In case `HTTPS` scheme is used, the CA certificate should be provided like this:
+
+```yaml
+spec:
+  scheme: HTTPS
+  tlsConfig:
+    ca:
+      secret:
+        name: <name-of-ca-bundle-secret>
+        key: bundle.crt
+```
+
+In case the component requires credentials when contacting its metrics endpoint, provide them like this:
+
+```yaml
+spec:
+  authorization:
+    credentials:
+      name: <name-of-secret-containing-credentials>
+      key: <data-keyin-secret>
+```
+
+If the component delegates authorization to the `kube-apiserver` of the shoot cluster, you can use the `shoot-access-prometheus-shoot` secret:
+
+```yaml
+spec:
+  authorization:
+    credentials:
+      name: shoot-access-prometheus-shoot
+      key: token
+  # in case the component's server certificate is signed by the cluster CA:
+  scheme: HTTPS
+  tlsConfig:
+    ca:
+      secret:
+        name: ca-bundle-08f7ac5d
+        key: bundle.crt
+```
+
+##### `ScrapeConfig`s
+
+If the component runs in the shoot cluster itself, metrics are scraped via the `kube-apiserver` proxy.
+In this case, Prometheus needs to authenticate itself with the API server.
+This can be done like this:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1alpha1
+kind: ScrapeConfig
+metadata:
+  labels:
+    prometheus: shoot
+  name: shoot-my-shoot-component
+  namespace: shoot--foo--bar
+spec:
+  authorization:
+    credentials:
+      key: token
+      name: shoot-access-prometheus-shoot
+  scheme: HTTPS
+  tlsConfig:
+    ca:
+      secret:
+        key: bundle.crt
+        name: ca-bundle-08f7ac5d
```

Comment:
Interesting that you consistently switched from the order `name`, `key` to `key`, `name` here.
Do we want to align the seed and shoot `ScrapeConfig`s to a consistent order?